src/librustc_target/spec/thumbv4t_nintendo_gba.rs

```diff
@@ -0,0 +1,80 @@
+//! Targets the Nintendo Game Boy Advance (GBA),
+//! a handheld game device from 2001.
+//!
+//! Please ping @Lokathor if changes are needed.
+//!
+//! The target profile assumes that you have the ARM binutils in your path (specifically the linker, `arm-none-eabi-ld`). They can be obtained for free for all major OSes from the ARM developer's website, and they may also be available in your system's package manager
+//!
+//! **Important:** This target profile **does not** specify a linker script or the ROM header. You'll still need to provide these yourself to construct a final  binary. Generally you'd do this with something like `-Clink-arg=-Tmy_script.ld` and `-Clink-arg=my_crt.o`.
+
+use crate::spec::{LinkerFlavor, PanicStrategy, RelocModel, Target, TargetOptions, TargetResult};
+
+pub fn target() -> TargetResult {
+    Ok(Target {
+        llvm_target: "thumbv4t-none-eabi".to_string(),
+        target_endian: "little".to_string(),
+        target_pointer_width: "32".to_string(),
+        target_c_int_width: "32".to_string(),
+        target_os: "none".to_string(),
+        target_env: "gba".to_string(),
+        target_vendor: "nintendo".to_string(),
+        arch: "arm".to_string(),
+        /* Data layout args are '-' separated:
+         * little endian
+         * stack is 64-bit aligned (EABI)
+         * pointers are 32-bit
+         * i64 must be 64-bit aligned (EABI)
+         * mangle names with ELF style
+         * native integers are 32-bit
+         * All other elements are default
+         */
+        data_layout: "e-S64-p:32:32-i64:64-m:e-n32".to_string(),
+        linker_flavor: LinkerFlavor::Ld,
+        options: TargetOptions {
+            linker: Some("arm-none-eabi-ld".to_string()),
+            linker_is_gnu: true,
+
+            // extra args passed to the external assembler
+            asm_args: vec!["-mcpu=arm7tdmi".to_string(), "-mthumb-interwork".to_string()],
+
+            cpu: "arm7tdmi".to_string(),
+
+            // minimum extra features, these cannot be disabled via -C
+            features: "+soft-float,+strict-align".to_string(),
+
+            executables: true,
+
+            relocation_model: RelocModel::Static,
+
+            //function_sections: bool,
+            //exe_suffix: String,
```

Comment:
```suggestion
```

Reply:
oh, right, i was going to ask. The docs for `function_sections` said that this defaults to `true`, but it seemed a little silly to me to give every function its own section. How does this section naming work exactly? Also, if I've been using `cargo-xbuild` and a custom target have I also ended up having this "on" already?

Basically, will the existing GBA linker scripts ([example](https://github.com/rust-console/gba/blob/master/linker.ld#L16)) keep working with `function_sections` on?

Reply:
They should keep working, we use similar ones for the embedded thumb targets. I highly recommend turning this on by default as it reduces executable sizes by letting the linker strip unused functions.